### PR TITLE
Functional tests - Add new test 'FO > User account > Order history > Order details > Request merchandise return'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'FO > User account > Order history > Order details > Request merchandise return'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

**Review by boubkerbribri — DISMISSED** [review]

**Review by SD1982 — DISMISSED** [review]

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      const result = await orderDetailsPage.isOrderReturnFormVisible(page);
+      await expect(result).to.be.true;
+    });
+
+    it('should create a merchandise return and check if merchandise return page is displayed', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'createMerchandiseReturn', baseContext);
+
+      await orderDetailsPage.requestMerchandiseReturn(page, 'Test merchandise returns');
+
+      const pageTitle = await foMerchandiseReturnsPage.getPageTitle(page);
```
Comment (SD1982) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts:188]: Type of `const pageTitle` is missing

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      await foOrderHistoryPage.goToDetailsPage(page);
+
+      const pageTitle: string = await orderDetailsPage.getPageTitle(page);
+      await expect(pageTitle).to.equal(orderDetailsPage.pageTitle);
+    });
+
+    it('should check the existence of order return form', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'isOrderReturnFormVisible', baseContext);
+
+      const result = await orderDetailsPage.isOrderReturnFormVisible(page);
```
Comment (SD1982) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts:179]: type is missing

Comment (SD1982) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts:184]: Type is missing

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts (hunk 1 above)
Comment (SD1982) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts:193]: Type is missing here too

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts:184]: Done thx :+1:

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
+// Import utils
+import helper from '@utils/helpers';
+import testContext from '@utils/testContext';
+import {Statuses} from '@data/demo/orderStatuses';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+import foOrderHistoryPage from '@pages/FO/myAccount/orderHistory';
+import invoicesPage from '@pages/BO/orders/invoices';
+import ordersPage from '@pages/BO/orders';
+import orderPageTabListBlock from '@pages/BO/orders/view/tabListBlock';
+import orderDetailsPage from '@pages/FO/myAccount/orderDetails';
+import foMerchandiseReturnsPage from '@pages/FO/myAccount/merchandiseReturns';
+
+// Import data
+import {DefaultCustomer} from '@data/demo/customer';
+import {PaymentMethods} from '@data/demo/paymentMethods';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts]:
```suggestion
import {Statuses} from '@data/demo/orderStatuses';
import {PaymentMethods} from '@data/demo/paymentMethods';
```

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts]: Done thx :)

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts (hunk 4 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts]: Done :+1:

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Could you remove type inference declarations ;) ?

File: tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+- Check the created return
+Post-condition:
+- Disable merchandise return
+ */
+describe('FO - Account - Order details : Request merchandise return', async () => {
+  let browserContext: BrowserContext;
+  let page: Page;
+  let orderReference: string;
+
+  const orderData: OrderCreator = {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/02_orderHistory/01_orderDetails/03_requestMerchandiseReturn.ts]: Could you use a `new OrderData`, plz ?

**Review by Progi1984 — COMMENTED** [review]
@nesrineabdmouleh A last feedback and it's ok. Sorry :pray:.

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Review by cfarhani06 — APPROVED** [review]

**Pull request merged**